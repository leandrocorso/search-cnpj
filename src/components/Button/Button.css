.btn {
    display: flex;
	align-items: center;
	justify-content: space-between;
	text-decoration: none;
	font-weight: 700;
	height: 5rem;
	padding: 1rem;
	
}

.btn svg {
	height: 1rem;
	width: 1rem;
}

.btn.default {
	color: #3cc;
	background-color: #fff;
}

.btn.default.disabled {
	color: rgba(51, 204, 204, .5);
	background-color: rgba(255, 255, 255, .5);
}

.btn.default svg{
	fill: rgba(51, 204, 204, .5);
}

.btn.primary {
	color: #fff;
	background-color: #3cc;
}

.btn.primary svg{
	fill: #fff;
}