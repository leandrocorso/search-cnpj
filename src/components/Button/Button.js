import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';

const ButtonStyle = styled(Link)`
	display: flex;
	align-items: center;
	justify-content: space-between;
	text-decoration: none;
	font-weight: 700;
	color: ${p => p.theme.themeDefault.button[p.color].color};
	height: 5rem;
	padding: 1rem;
	background-color: ${p => p.theme.themeDefault.button[p.color].background};
`;

const ButtonIcon = styled(Icon)`
	fill: ${p => p.theme.themeDefault.button[p.color].color};
	height: 1rem;
	width: 1rem;
`;

const Button = ({ children, ...rest }) => (
	<React.Fragment>
		<ButtonStyle to={rest.to} icon={rest.icon} color={rest.color} {...rest} className="btn">
			{children}
			{rest.icon && <ButtonIcon src={rest.icon} color={rest.color}/>}
		</ButtonStyle>
	</React.Fragment>
);

export default Button;
