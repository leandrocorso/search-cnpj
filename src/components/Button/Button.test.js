import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('Testing Button component', () => {
    
    const wrapper = shallow(<Button>Text button</Button>);
    const element = wrapper.find('.btn');
    
    test('Button exists', () => {
        expect(element);
    });

    test('Should have a text', () => {
        expect(element).toHaveText();
    });

    test('Should have a route', () => {
        expect(element).toHaveProp('to');
    });

    test('Should have a icon', () => {
        expect(element).toHaveProp('icon');
    });

    test('Should have a color', () => {
        expect(element).toHaveProp('color');
    });

    test('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

})