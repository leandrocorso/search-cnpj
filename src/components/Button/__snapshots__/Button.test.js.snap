// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Should render correctly 1`] = `
<Fragment>
  <Styled(Link)
    className="btn"
  >
    Text button
  </Styled(Link)>
</Fragment>
`;
